core/offchain.go

```diff
@@ -130,24 +131,34 @@ func (bc *BlockChain) CommitOffChainData(
 		}
 	}
 
-	//// Writing beacon chain cross links
+	// Writing beacon chain cross links
 	if header.ShardID() == shard.BeaconChainShardID &&
 		bc.chainConfig.IsCrossLink(block.Epoch()) &&
 		len(header.CrossLinks()) > 0 {
 		crossLinks := &types.CrossLinks{}
-		err = rlp.DecodeBytes(header.CrossLinks(), crossLinks)
-		if err != nil {
-			header.Logger(utils.Logger()).Warn().Err(err).Msg("[insertChain/crosslinks] cannot parse cross links")
+		if err := rlp.DecodeBytes(
+			header.CrossLinks(), crossLinks,
+		); err != nil {
+			header.Logger(utils.Logger()).
+				Warn().Err(err).
+				Msg("[insertChain/crosslinks] cannot parse cross links")
 			return NonStatTy, err
 		}
 		if !crossLinks.IsSorted() {
-			header.Logger(utils.Logger()).Warn().Err(err).Msg("[insertChain/crosslinks] cross links are not sorted")
+			header.Logger(utils.Logger()).
+				Warn().Err(err).
+				Msg("[insertChain/crosslinks] cross links are not sorted")
 			return NonStatTy, errors.New("proposed cross links are not sorted")
 		}
 		for _, crossLink := range *crossLinks {
 			// Process crosslink
-			if err := bc.WriteCrossLinks(batch, types.CrossLinks{crossLink}); err == nil {
-				utils.Logger().Info().Uint64("blockNum", crossLink.BlockNum()).Uint32("shardID", crossLink.ShardID()).Msg("[insertChain/crosslinks] Cross Link Added to Beaconchain")
+			if err := bc.WriteCrossLinks(
+				batch, types.CrossLinks{crossLink},
+			); err == nil {
```

Comment:
is this log info or debug? we have flooded the log already.

Reply:
it is info. This log will be O(1) as hoorays